feature/experimental/bgp/ate_tests/bgp_long_lived_graceful_restart/bgp_long_lived_graceful_restart_test.go

```diff
@@ -0,0 +1,1339 @@
+		t.Fatalf("Unsupported vendor %s for deviation 'UseVendorNativeACLConfiguration'", d.Vendor())
+	}
+}
+
+// Helper function to replicate AdmitAllACL() configs in native model,
+// then craft a gNMI set Request to update the changes.
+func configAdmitAllACLNative(t testing.TB, d *ondatra.DUTDevice, name string) {
+	t.Helper()
+	switch d.Vendor() {
+	case ondatra.NOKIA:
+		gpbDelRequest := &gpb.SetRequest{
+			Prefix: &gpb.Path{
+				Origin: "srl",
+			},
+			Delete: []*gpb.Path{
+				{
+					Elem: []*gpb.PathElem{
+						{Name: "acl"},
+						{Name: "ipv4-filter", Key: map[string]string{"name": name}},
+						{Name: "entry", Key: map[string]string{"sequence-id": "10"}},
+					},
+				},
+				{
+					Elem: []*gpb.PathElem{
+						{Name: "acl"},
+						{Name: "ipv4-filter", Key: map[string]string{"name": name}},
+						{Name: "entry", Key: map[string]string{"sequence-id": "20"}},
+					},
+				},
+			},
+		}
+		gnmiClient := d.RawAPIs().GNMI().Default(t)
+		if _, err := gnmiClient.Set(context.Background(), gpbDelRequest); err != nil {
+			t.Fatalf("Unexpected error removing SRL ACL: %v", err)
+		}
+	default:
+		t.Fatalf("Unsupported vendor %s for deviation 'UseVendorNativeACLConfiguration'", d.Vendor())
+	}
+}
+
+// Helper function to replicate configACLInterface in native model.
+// Set ACL at interface ingress,
+// then craft a gNMI set Request to update the changes.
+func configACLInterfaceNative(t *testing.T, d *ondatra.DUTDevice, ifName string) {
+	t.Helper()
+	switch d.Vendor() {
+	case ondatra.NOKIA:
+		var interfaceAclVal = []any{
+			map[string]any{
+				"ipv4-filter": []any{
+					aclName,
+				},
+			},
+		}
+		interfaceAclUpdate, err := json.Marshal(interfaceAclVal)
+		if err != nil {
+			t.Fatalf("Error with json Marshal: %v", err)
+		}
+		gpbSetRequest := &gpb.SetRequest{
+			Prefix: &gpb.Path{
+				Origin: "srl",
+			},
+			Update: []*gpb.Update{
+				{
+					Path: &gpb.Path{
+						Elem: []*gpb.PathElem{
+							{Name: "interface", Key: map[string]string{"name": ifName}},
+							{Name: "subinterface", Key: map[string]string{"index": "0"}},
+							{Name: "acl"},
+							{Name: "input"},
+						},
+					},
+					Val: &gpb.TypedValue{
+						Value: &gpb.TypedValue_JsonIetfVal{
+							JsonIetfVal: interfaceAclUpdate,
+						},
+					},
+				},
+			},
+		}
+		gnmiClient := d.RawAPIs().GNMI().Default(t)
+		if _, err := gnmiClient.Set(context.Background(), gpbSetRequest); err != nil {
+			t.Fatalf("Unexpected error configuring interface ACL: %v", err)
+		}
+	default:
+		t.Fatalf("Unsupported vendor %s for deviation 'UseVendorNativeACLConfiguration'", d.Vendor())
+	}
+}
+
+func juniperCLI() string {
+	var proto string = "bgp"
+	return fmt.Sprintf(`
+	protocols {
+		%s {
+			graceful-restart {
+				long-lived {
+					receiver {
+						disable;
+					}
+				}
+			}
+		}
+	}
+	`, proto)
+}
+
+func removeATENewPeers(t *testing.T, ate *ondatra.ATEDevice, topo *ondatra.ATETopology, intfList []*ondatra.Interface, bgpPeers []*ixnet.BGP) {
```

Comment:
remove ate and intfList parameters -- not used

Reply:
updated.